node/src/actors/chain_manager/mod.rs

```diff
@@ -715,6 +798,11 @@ impl ChainManager {
         })
         .into_actor(self)
         .and_then(move |(), act, _ctx| {
+            // Check if we already received this vote (again, because this future can be executed
+            // by multiple tasks concurrently)
+            if act.chain_state.superblock_state.contains(&superblock_vote) {
+                return actix::fut::err(());
```

Comment:
In lines 710 and 725 we returns Ok, now we have similar cases (789, 804) but we return an Error here, is it make sense?

Reply:
It is not the best way to do it, but it seems it is used for signalling that the vote shouldn't be broadcasted again.